### ISSUE TITLE
While on Live Share, docstring quotes always auto-inserted at host's cursor

### ISSUE THREAD

**Issue by aely-aronoff** [microsoft/pylance-release#1583]
TL;DR: On Live Share, if a remote machine types three quotes, a matching set of three quotes appears at the host machine's terminal instead of the remote machine.

First reported in: https://github.com/microsoft/vscode/issues/127910#issue-936117695
Check there for additional details.

## Environment data

Host machine:
-   VS Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:28:07.755Z)
-   Extension version (available under the Extensions sidebar): 2021.6.944021595
-   OS and version: Windows_NT x64 10.0.19042
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.1 64-bit
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
-   Relevant/affected Python-related VS Code extensions and their versions: vsliveshare v1.0.4583
-   Value of the `python.languageServer` setting: default, which is I think pylance

Remote machine:
-   VS Code version: Code 1.58.2
-   Extension version (available under the Extensions sidebar): 2021.6.944021595
-   OS and version: 11.1 macOS Big Sur
-   Relevant/affected Python-related VS Code extensions and their versions: vsliveshare v1.0.4583

## Expected behaviour

While using Live Share, if one of the participants types quote (`"`) three times to create a docstring, the other three quotes should appear at their cursor. 

## Actual behaviour

The second set of three quotes appears at the host's cursor. See full description in: https://github.com/microsoft/vscode/issues/127910#issue-936117695.

The issue was moved to the live-share repo: https://github.com/MicrosoftDocs/live-share/issues/4353.

## Steps to reproduce:

1. With VS Code on a host machine and remote machine, start a Live Share session and connect with remote.
2. Have both machines open the same Python file.
3. On host machine, type three quotes. Docstring should insert correctly.
4. On remote machine, type three quotes. 
5. Expected behavior is that three more quotes appear at remote machine's cursor, but instead they appear at host's cursor.

**Comment by jakebailey** [comment]
> I suspect this is an issue with the Python extension - they're probably listening for the triple quotes being typed, and then inserting the closing triple quotes. They should only be doing that if the opening triple quotes were typed by the local user.

This one is interesting; I don't know if the event stream indicates who is the one typing...

**Comment by jakebailey** [comment]
I should be able to fix this by not using the cursor at all, but instead use the location of the typing.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.7.5, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202175-22-july-2021

**Comment by aely-aronoff** [comment]
@jakebailey, good news and bad news.

Good news: The original bug was fixed. Docstring quotes are now inserted at the remote cursor instead of at the host.
Bad news: There's now a new bug where the host's cursor gets moved to the remote's cursor when the docstring quotes are inserted.

Would you like me to submit a new issue for that?

**Comment by jakebailey** [comment]
Ugh, so close. Do file a new issue.

I'm a bit at a loss of how to prevent that. VS Code text edits are really finnicky when they are at the cursor. Best would be for VS Code to support multi-character auto-close itself, rather than me implementing it.

**Comment by aely-aronoff** [comment]
Opened ticket #1592
